server/src/main/java/org/elasticsearch/common/bytes/ReleasableBytesReference.java

```diff
@@ -26,9 +26,17 @@
 public final class ReleasableBytesReference implements RefCounted, Releasable, BytesReference {
 
     public static final Releasable NO_OP = () -> {};
+
+    private static final ReleasableBytesReference EMPTY = new ReleasableBytesReference(BytesArray.EMPTY, NO_OP);
+
     private final BytesReference delegate;
     private final AbstractRefCounted refCounted;
 
+    public static ReleasableBytesReference empty() {
+        EMPTY.incRef();
```

Comment:
Do we actually need to keep track of the ref count for empty? Seems like it wouldn't be necessary since the releasable is a no-op.

Reply:
True, but `ReleasableBytesReference` is final and `refCounted` is an `AbstractRefCounted` as well so we can't override anything here. Plus, it's kind of nice to retain the assertions on the ref count I suppose even if there's a small/trivial cost to it?